src/main/java/com/minecolonies/coremod/commands/AbstractCommandParser.java

```diff
@@ -0,0 +1,649 @@
+        if (!menu.getMenuType().isNavigationMenu())
+        {
+            final ActionMenu actionMenu = (ActionMenu) menu;
+            for (final ActionArgument actionArgument : actionMenu.getActionArgumentList())
+            {
+                if (usedThisActionArgumentName(parsedHolders, actionArgument))
+                {
+                    continue;
+                }
+                map.put(actionArgument.getName().toLowerCase(Locale.ROOT) + ":", new ActionMenuHolder(treeNode, actionArgument));
+            }
+        }
+        return map;
+    }
+
+    private boolean usedThisActionArgumentName(@NotNull final List<ActionMenuHolder> parsedHolders, @NotNull final ActionArgument actionArgument)
+    {
+        for (final ActionMenuHolder actionMenuHolder : parsedHolders)
+        {
+            if (actionMenuHolder.getActionArgument().getName().equals(actionArgument.getName()))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    @NotNull
+    @Override
+    public List<String> getTabCompletions(
+                                           @NotNull final MinecraftServer server,
+                                           @NotNull final ICommandSender sender,
+                                           @NotNull final String[] args,
+                                           @Nullable final BlockPos pos)
+    {
+        final ModuleContext moduleContext = getModuleContext();
+        return getTabCompletions(server, sender, args, pos, moduleContext);
+    }
+
+    @NotNull
+    protected List<String> getTabCompletions(
+                                           @NotNull final MinecraftServer server,
+                                           @NotNull final ICommandSender sender,
+                                           @NotNull final String[] args,
+                                           @Nullable final BlockPos pos,
+                                           @NotNull final ModuleContext moduleContext)
+    {
+        final ParsingResult parsingResult = getTabCompletionsAndParsingHolders(getRoot(), server, sender, args, pos, moduleContext);
+        return parsingResult.getTabCompletions();
+    }
+
+    @NotNull
+    private ParsingResult getTabCompletionsAndParsingHolders(
+                                           @NotNull final TreeNode<IMenu> treeNode,
+                                           @NotNull final MinecraftServer server,
+                                           @NotNull final ICommandSender sender,
+                                           @NotNull final String[] args,
+                                           @Nullable final BlockPos pos,
+                                           @NotNull final ModuleContext moduleContext)
+    {
+        if (treeNode.getData().getMenuType().isNavigationMenu())
+        {
+            final Map<String, TreeNode<IMenu>> childs = getNavigationCommands(treeNode);
+            if (0 == args.length) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)